### ISSUE TITLE
Feature: support multiple parameters in `.startsWith()`

### ISSUE THREAD

**Issue by jansepke** [colinhacks/zod#3683]
Currently it quite verbose to define a string that can start with multiple different prefixes:

```js
const schema = z
  .string()
  .startsWith('foo_')
  .or(z.string().startsWith('bar_'))
  .or(z.string().startsWith('baz_'));
```

it would be great if `.startsWith()` would support an array of strings like
```js
const schema = z
  .string()
  .startsWith(['foo_', 'bar_', 'baz_']);
```

maybe I am also missing an alternative solution for the problem.

**Comment by tristinDLC** [comment]
You can do it cleaner with regex:

```javascript
const schema = z
    .string()
    .regex(/^foo_|bar_|baz_/)
```

I'll admit that using `.startsWith()` is a bit better because you can quickly see what it's intended to do, but the regex pattern will functionally cover your use-case.

**Comment by jansepke** [comment]
@tristinDLC yes I know, but using regex always hides the intention. But I would also close this issue if this would add too much complexity for not enough value to zod.

**Comment by Kumar06Lav** [comment]
I've submitted a PR to add support for multiple parameters in the .startsWith() and .endsWith() methods for string validation.

**Comment by jansepke** [comment]
The issue is still relevant with the latest version of Zod

**Comment by jansepke** [comment]
Not stale

**Comment by jansepke** [comment]
Not stale

**Comment by colinhacks** [comment]
Closing this out — declined in #5935. `.startsWith()` taking an array reads fine on its own, but `.endsWith()` and `.includes()` wouldn't follow, so the string API would be inconsistent, and it widens `issue.prefix` for everyone with a custom error map.

`z.string().regex(/^(foo_|bar_|baz_)/)` is the way — keep the group, since `/^foo_|bar_/` anchors the first branch only. Thanks for keeping this one alive 👍
